Modules/posixmodule.c

```diff
@@ -5489,8 +5463,15 @@ os_posix_spawn_impl(PyObject *module, path_t *path, PyObject *argv,
     attrp = &attr;
 
     _Py_BEGIN_SUPPRESS_IPH
-    err_code = posix_spawn(&pid, path->narrow,
+    if (use_posix_spawnp) {
```

Comment:
This code will fail with a compilation error if posix_spawn() is available (HAVE_POSIX_SPAWN defined) but not posix_spawnp() (HAVE_POSIX_SPAWNP not defined).

I suggest to write:

```
    _Py_BEGIN_SUPPRESS_IPH
#ifdef HAVE_POSIX_SPAWNP
    if (use_posix_spawnp) {
        err_code = posix_spawnpx(&pid, path->narrow,
                           file_actionsp, attrp, argvlist, envlist);
    }
    else
#endif /* HAVE_POSIX_SPAWNP */
    {
        err_code = posix_spawn(&pid, path->narrow,
                           file_actionsp, attrp, argvlist, envlist);
    }
    _Py_END_SUPPRESS_IPH
```

Reply:
I pushed a commit to fix it directly.